### PR TITLE
EZP-27142: Update migration guide for 1.11 and legacy bridge

### PR DESCRIPTION
**Pull request by andrerom**
### What is this?

This changes current migration documentation [to also support usage of legacy bridge](https://ez.no/Blog/eZ-Platform-Enterprise-Now-Officially-Supports-Use-with-Legacy-Bridge), with this, installation of eZ Platform can be done three ways:
- Clean install for new projects, no legacy
- migration, no legacy => you should migrate all content (XmlText, Flow) during migration steps as Platform UI does not support them*
- migration, with legacy => you can continue to use legacy admin and can postpone content migration, meaning you won't be able to fully use Platform UI*

\* _This can theoretically change a bit with v2 if community contributes support simplified XmlText and Flow editing, further relaxing how instant content migration needs to be done as long as you are ok with using community supported packages with gradually less features in Platform then their newer native and activly developed counterparts (RichText, EE Studio Landing Page)_

#### Why no doc on clean install with legacy?

As setting up and handling projects using platform + legacy is far more complex then pure Platform install, we _(eZ Systems)_ neither recommend nor support this option. However documentation for this can be contributed by community, we just need to create own section for community documentation and better clarify what is support and not with eZ's Enterprise Support offering.

#### Todo:
- [x] Pending QA in the next week(s), review welcome
- [ ] When merging up to 1.12 there is a new SQL file for password improvements that should be added: https://github.com/ezsystems/ezpublish-kernel/blob/master/data/update/mysql/dbupdate-6.11.0-to-6.12.0.sql
- [x] Add doc to reflect upcoming support in Legacy Bridge for natively handling symlinking settings and design files to simplify/solidify migration steps: https://github.com/ezsystems/LegacyBridge/pull/107
- [x] Get update sql into a patch release of 1.11, it's currently missing: https://github.com/ezsystems/ezpublish-kernel/compare/v6.11.0...6.11

### CODE REVIEW

File: docs/migrating/migrating_from_ez_publish_platform.md [hunk 1]
```diff
@@ -144,27 +179,27 @@ Add the following new bundle to your new kernel file, `<new-ez-root>/app/AppKern
...
-For migrating content from the XmlText format to the new RichText format, a migration script exists, execute the following from &lt;new-ez-root&gt;:
+**If** you are ready to migrate your eZ Publish XMLText content to the eZ Platform RichText format and start using pure eZ Platform setup, you can use a script to migrate content from the XmlText format to the new RichText format. Execute the following from &lt;new-ez-root&gt;:
 
 `php app/console ezxmltext:convert-to-richtext -v`
 
-The migration script is currently in beta, which is why command example is suggested with verbose flag. Feedback on how it works and on how we can improve it is welcome [on the repository](https://github.com/ezsystems/ezplatform-xmltext-fieldtype).
+The command example suggests using the verbose flag. This is optional, but recommended as we are actively gathering feedback on how it works with older eZ Publish content, and on how we can improve it both [via issues in the repository](https://github.com/ezsystems/ezplatform-xmltext-fieldtype), and on Slack.
 
-##### 3.3. Migrate Page field to Landing Page (eZ Enterprise only)
+###### 3.2.2 Migrate Page field to Landing Page (eZ Platform Enterprise only)
```
Comment (andrerom) [inline docs/migrating/migrating_from_ez_publish_platform.md]: `Platform` part might be wrong here, I'll let @DominikaK give input on what should be used here.

**Comment by andrerom** [comment]
This doc should not more or less be ready for QA using:https://github.com/ezsystems/LegacyBridge/releases/tag/v1.4.0-beta1.

However I'll do a second pass in the next few days, and its open for another round for text proofing before being sent to QA together with 1.12-beta release.

File: docs/migrating/migrating_from_ez_publish_platform.md [hunk 2]
```diff
@@ -144,27 +179,32 @@ Add the following new bundle to your new kernel file, `<new-ez-root>/app/AppKern
...
+
+**If** you use Page field (ezflow) and an eZ Enterprise subscription, and are ready to migrate your eZ Publish Flow content to the eZ Enterprise Landing Page format. You can use a script to migrate your Page content to Landing Page, to start using a pure eZ Enterprise setup. See [Migrating legacy Page field (ezflow) to Landing Page (Enterprise)](#migrating-legacy-page-field-ezflow-to-landing-page-enterprise) for more information.
 
-##### 3.3. Migrate Page field to Landing Page (eZ Enterprise only)
+###### 3.2.3 Add Date Based Publisher schema (eZ Enterprise only)
 
-If you have Page field (ezflow) content and an eZ Enterprise subscription, you can use a script to migrate your Page content to eZ Enterprise Landing Page. See [Migrating legacy Page field (ezflow) to Landing Page (Enterprise)](#migrating-legacy-page-field-ezflow-to-landing-page-enterprise) for more information.
+For date based publisher feature, there is an additional table, you can import it to your database using the following sql file:
+`<new-ez-root>/vendor/ezsystems/date-based-publisher/bundle/Resources/install/datebasedpublisher_scheduled_version.sql`
```
Comment (andrerom) [inline docs/migrating/migrating_from_ez_publish_platform.md]: I'll need dev & doc team to own this section, what else needs to be added here for migration to EE install with Studo?

Customer that uncovered this part was missing now also is reporting missing table from: `vendor/ezsystems/ezstudio-form-builder/bundle/Resources/install/form_builder.sql`. What else need to be mentioned here? Is there one single place in installer we can link to / always be able to verify against to make sure we have all? /cc @lserwatka

File: docs/migrating/migrating_from_ez_publish_platform.md [hunk 3]
```diff
@@ -108,23 +108,58 @@ To move over your own custom configurations, follow the conventions below and ma
...
+
+1. The same goes for the `<new-ez-root>/ezpublish_legacy/var/[<site>/]storage` folder. However, as it is typically not versioned in git, there's no predefined convention for this. If you create a folder within your project structure for symlinking into this folder, as opposed to a mount somewhere else, make sure to mark this folder as ignored by git once it and the corresponding `.keep` file have been added to your checkout. The example below will assume `<new-ez-root>/src/legacy_files/storage` was created for this purpose, if you opt for something else just adjust the instructions.
+
+###### 2.5.2 Upgrade the legacy distribution files
+
+The easiest way to upgrade the distribution files is to copy the directories that contain site-specific files from the existing 5.4 installation into `/<ezplatform>/ezpublish_legacy`. Make sure you copy the following directories:
+
+- `<old-ez-root>/ezpublish_legacy/design/<your_designs>` => `<new-ez-root>/src/legacy_files/design/<your_designs>`
+    - *Do NOT include built-in designs: admin, base, standard or admin2*
+- `<old-ez-root>/ezpublish_legacy/design/<your_designs>` => `<new-ez-root>/src/legacy_files/design/<your_designs>`
```
[the remark this answers is not in the document]
Comment (andrerom) [reply docs/migrating/migrating_from_ez_publish_platform.md]: fixed in 8fa33a5

File: docs/migrating/migrating_from_ez_publish_platform.md [hunk 4]
```diff
@@ -1,28 +1,23 @@
...
-!!! note
-
-    Instructions for upgrading from eZ Publish to eZ Platform and eZ Enterprise are in preview starting release [16.02](../releases/ez_platform_16.02_release_notes.md). The status of the upgrade is:
-
-    - eZ Platform: **XmlText to RichText migration**: *In Beta, and described below.*
-    - eZ Enterprise: **Flow to Landing Page migration**: *Scheduled for beta version with 16.04.*
+        - [Field Types reference](../guide/field_type_reference.md) for overview of Field Types that do and don't exist in eZ Platform
+        - eZ Platform RichText Field Type capabilities, currently not covering [Custom Tags](https://jira.ez.no/browse/EZP-25357)
+        - Symfony 2.8, this is also the case on later 5.4.x versions, but not the first ones including 2014.11
+        - API changes. While we have a strict backwards compatibility focus, some deprecated API features were removed and some changes where done to internal parts of the system. See [ezpublish-kernel:doc/bc/changes-6.0.md](https://github.com/ezsystems/ezpublish-kernel/blob/v6.7.0/doc/bc/changes-6.0.md)
```
Comment (natanael89) [inline docs/migrating/migrating_from_ez_publish_platform.md]: Nitpick: "were done" instead of "where done"

**Comment by andrerom** [comment]
@micszo I agree I think, it will be simpler to follow this if we merge and then do smaller changes to improve it. This already vastly improved the doc on migration as it without this is more or less broken.

Any further review notes before we merge?

**Comment by andrerom** [comment]
@DominikaK Do you update 1.12 and up with the added sql file on migrations? It can be added in `3.1. Execute update SQL` section

**Comment by DominikaK** [comment]
@andrerom, so for 1.12 I need both `dbupdate-5.4.0-to-6.11.0.sql` and `dbupdate-6.11.0-to-6.12.0.sql`, right?

**Comment by andrerom** [comment]
@DominikaK yes, for 1.13LTS I'll make a single file again, but we don't do that on every single FT release.

**Comment by DominikaK** [comment]
Thanks, added in https://github.com/ezsystems/developer-documentation/commit/3656faab32286178121c1216301f08e413cb3268

**Pull request merged**